src/main/kotlin/it/pagopa/wallet/cdc/PaymentWalletsLogEventsStream.kt

```diff
@@ -44,22 +51,47 @@ class PaymentWalletsLogEventsStream(
                                 Aggregation.project(changeStreamOptionsConfig.project)
                             )
                         )
-                        .resumeAt(Instant.now())
+                        .resumeAt(redisResumePolicyService.getResumeTimestamp())
                         .build(),
                     BsonDocument::class.java
                 )
-                .flatMap {
-                    Mono.defer {
-                            walletPaymentCDCEventDispatcherService.dispatchEvent(
-                                it.raw?.fullDocument?.toBsonDocument()
-                            )
-                        }
-                        .onErrorResume {
-                            logger.error("Error during event handling : ", it)
-                            Mono.empty<BsonDocument>()
-                        }
-                }
+                // Process the elements of the Flux
+                .flatMap { processEvent(it) }
+                // Save resume token every n emitted elements
+                .index()
+                .flatMap { saveToken(it) }
+                .doOnError { logger.error("Error listening to change stream: ", it) }
 
         return flux
     }
+
+    private fun processEvent(event: ChangeStreamEvent<BsonDocument>): Mono<BsonDocument> {
+        return Mono.defer {
+                walletPaymentCDCEventDispatcherService.dispatchEvent(
+                    event.raw?.fullDocument?.toBsonDocument()
+                )
+            }
+            .onErrorResume {
+                logger.error("Error during event handling : ", it)
+                Mono.empty()
+            }
+    }
+
+    private fun saveToken(tuple: Tuple2<Long, BsonDocument>): Mono<BsonDocument> {
```

Comment:
tuple as a method parameter could be difficult to read, what is the long of the tuple? why not use a parameter with a name that refers to the value of this long?

Reply:
Renamed in changeEventFluxIndex in [0312257](https://github.com/pagopa/pagopa-payment-wallet-cdc-service/pull/11/commits/03122570f12cdd4260bf695fdbbe9fd09d4fe114). Do you think is easier to read now or should be renamed in something else?

Reply:
Applied some refactoring suggested by @pietro-tota here [6f724d6](https://github.com/pagopa/pagopa-payment-wallet-cdc-service/pull/11/commits/6f724d6a62d4c67307cc64fbf807e7eb10199470)